core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalSystem.java

```diff
@@ -137,6 +145,32 @@ public CatchupFuture catchup(Map<String, Long> journalSequenceNumbers) throws IO
     return CatchupFuture.allOf(futures);
   }
 
+  @Override
+  public boolean waitForInitialReplay() {
+    long start = System.currentTimeMillis();
+    try {
+      CommonUtils.waitFor("journal replay to finish catching up", () -> {
+        for (UfsJournal journal : mJournals.values()) {
+          UfsJournalCheckpointThread.ReplayState replayState = journal.getReplayState();
+          if (replayState != UfsJournalCheckpointThread.ReplayState.REPLAY_DONE) {
+            // keep flushing entries
+            return false;
+          }
+        }
+        return true;
+      }, WaitForOptions.defaults().setTimeoutMs(
+          (int) ServerConfiguration.getMs(PropertyKey.MASTER_JOURNAL_MAX_INITIAL_REPLAY_TIME))
+          .setInterval(30 * Constants.SECOND_MS));
```

Comment:
Should this be more frequent for faster startup?

Reply:
Yeah, i don't think it is any harmful if do it at 5-second interval since our shutdown quiet wait time is 5 second.